beacon-chain/sync/subscriber.go

```diff
@@ -331,6 +301,76 @@ func (r *Service) subscribeDynamic(topicFormat string, determineSubsLen func() i
 	}()
 }
 
+// revalidate that our currently connected subnets are valid.
+func (r *Service) reValidateSubscriptions(subscriptions map[uint64]*pubsub.Subscription,
+	wantedSubs []uint64, topicFormat string) {
+	for k, v := range subscriptions {
+		var wanted bool
+		for _, idx := range wantedSubs {
+			if k == idx {
+				wanted = true
+				break
+			}
+		}
+		if !wanted && v != nil {
+			v.Cancel()
+			r.p2p.PubSub().UnregisterTopicValidator(fmt.Sprintf(topicFormat, k))
+			delete(subscriptions, k)
+		}
+	}
+}
+
+// subscribe missing subnets for our aggregators.
+func (r *Service) subscribeMissingSubnet(subscriptions map[uint64]*pubsub.Subscription, idx uint64,
+	base proto.Message, digest [4]byte, validate pubsub.Validator, handle subHandler) {
+	// do not subscribe if we have no peers in the same
+	// subnet
+	topic := p2p.GossipTypeMapping[reflect.TypeOf(&pb.Attestation{})]
+	subnetTopic := fmt.Sprintf(topic, digest, idx)
+	if !r.validPeersExist(subnetTopic, idx) {
```

Comment:
maybe we should  FindPeersWithSubnet even if we have valid peers already that way we can add more peers if we find ones

Reply:
i see that now that we do it on `lookupAttesterSubnets`